File: druthvik/tutorial/server/src/controllers/studentmanagement/studentsranking/model/student.ts
import { Subject } from './subject';
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  OneToMany,
  OneToOne,
  JoinColumn,
} from 'typeorm';

@Entity()
export class Student {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @Column({ nullable: true })
  avgMarks: number;

  @OneToMany((_type) => Subject, (subject: Subject) => subject.student)
  subject: Array<Subject>;
}

File: harsha/School-Teacher-Student/server/src/main/java/com/example/manytomanymapping/models/Student.java
package com.example.manytomanymapping.models;

import java.util.Date;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import org.hibernate.annotations.CreationTimestamp;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "students")
public class Student {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	@Column(name = "student_id")
	private int studentId;
	
	@Column(name = "student_name")
	private String studentName;
	private String gender;

	@JsonFormat(pattern = "YYYY-MM-dd")
	private Date dateOfBirth;

	private String address;
	private String email;
	private long phoneNo;

	@Column(name = "created_date", nullable = false, updatable = false)
	@CreationTimestamp
	private Date createdDate;

	@JsonIgnoreProperties({"students","teachers"})
	@OneToOne(cascade = CascadeType.MERGE)
	@JoinColumn(name = "school_id")
	private School school;
}